lib/ui/views/delete_account/delete_account_bloc.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:nowu/router.gr.dart';
+import 'package:nowu/ui/views/delete_account/delete_account_state.dart';
+
+import '../../../services/auth.dart';
+import '../../../services/user_service.dart';
+
+class DeleteAccountBloc extends Cubit<DeleteAccountState> {
+  AuthenticationService _authenticationService;
+  StackRouter _appRouter;
+  UserService _userService;
+
+  DeleteAccountBloc({
+    required AuthenticationService authenticationService,
+    required StackRouter appRouter,
+    required UserService userService,
+  })  : _authenticationService = authenticationService,
+        _appRouter = appRouter,
+        _userService = userService,
+        super(const DeleteAccountState());
+
+  void updateInputName(String name) {
+    emit(
+      state.copyWith(
+        name: name,
+        isNameValid: name == _userService.currentUser?.name,
+      ),
+    );
+  }
+
+  Future<void> deleteAccount() async {
+    if (!state.isNameValid) return;
+
+    await _userService.deleteUser();
+    await _authenticationService.logout();
+    _appRouter.pushAndPopUntil(const LoginRoute(), predicate: (r) => false);
```

Comment:
Im not sure navigating the user to the login route is the right path here. If they are deleting their account Im guessing its more likely they don't want an account rather than they want to log into a different one. I could be wrong though... (e.g. maybe if the user wanted to change email or something but I think we can offer a different flow for that)

If you think that's right then we should probably leave them on the more menu page but have some kind of notification to show that the delete was successful... We might need to chat to design folks about that though.